### PR TITLE
Check if video choice is valid

### PR DESCRIPTION
**Pull request by mihaiolteanu**
Check if the video choice is a valid digit or if it's in the range of available choices.

### CODE REVIEW

File: instantmusic-0.1/bin/instantmusic [hunk 1]
```diff
@@ -114,7 +114,7 @@ def query_and_download(search, has_prompts=True, is_quiet=False):
     # We only ask the user which one they want if prompts are on, of course
     if has_prompts and not is_quiet:
         choice = ''
-        while choice.strip() == '':
+        while not(choice.isdigit()) or not(0 <= int(choice) < len(available)):
```
Comment (yask123) [inline instantmusic-0.1/bin/instantmusic]: We should also prompt user to enter valid input

Comment (mihaiolteanu) [inline instantmusic-0.1/bin/instantmusic]: I actually think the current solution would be nicer. Reprinting the "Pick one: " one prompt should be enough as a hint to the user that something went wrong. 
But we could do something like:

```
choice = raw_input('Pick one: ')
while not(choice.isdigit()) or not(0 <= int(choice) < len(available)):
    print("Oups, that was wrong. Try again!")
    choice = raw_input('Pick one: ')
```

Comment (yask123) [inline instantmusic-0.1/bin/instantmusic]:
```
while not(choice.isdigit()) or not(0 <= int(choice) < len(available)):
    print("Oups, that was wrong. Try again!")
```

We come inside the loop if the input is valid., then why respond by `print("Oups, that was wrong. Try again!")` ?

Comment (mihaiolteanu) [inline instantmusic-0.1/bin/instantmusic]: Nope. We come inside the loop if the input is **in**valid.

Comment (yask123) [inline instantmusic-0.1/bin/instantmusic]: Input is good if (0 <= int(choice) < len(available).
not  (0 <= int(choice) < len(available) -> makes it false
Outside of this while condition we have `not` -> makes it true

Comment (mihaiolteanu) [inline instantmusic-0.1/bin/instantmusic]: No, the first `not` is only for the `choice.isdigit()` and not for the whole `while`:

<pre>while not<b>(</b>choice.isdigit()<b>)</b> or not<b>(</b>0 <= int(choice) < len(available)<b>)</b>:</pre>

Comment (yask123) [inline instantmusic-0.1/bin/instantmusic]: Ah! sorry! I thought it was for whole :stuck_out_tongue:  I need my coffee!

**Comment by yask123** [comment]
Yea, this will be a better response:
```
 print("Oups, that was wrong. Try again!")
    choice = raw_input('Pick one: ')
```

**Comment by mihaiolteanu** [comment]
I've pushed the changes.

**Comment by yask123** [comment]
Thanks! :100:

**Pull request merged**